source/main/providers/citeproc-provider.ts

```diff
@@ -494,3 +456,9 @@ class CiteprocProvider {
 }
 
 module.exports = new CiteprocProvider()
+
+global.citeproc = module.exports
+
+// Listen for synchronous citation messages from the renderer
+// Citeproc calls (either single citation or a whole cluster)
+IpcModule.registerMain<IpcCiteService>(module.exports)
```

Comment:
Something regarding this implementation bugs me, but I can't really pinpoint it … maybe it's that it's very intransparent and not much verbose …

(+ here are all trailing newlines missing)

Reply:
I thought the short and simple code was actually an advantage ;-)

Reply:
Yeah, short code is always good, but I realized in the past three years that two or three more lines sometimes helps make the code more readable – and especially since we're finally talking about compiled code, these lines would get shortened either way :)